docs/xxenableextendedhcr.md

```diff
@@ -0,0 +1,52 @@
+﻿<!--
+* Copyright (c) 2017, 2024 IBM Corp. and others
+*
+* This program and the accompanying materials are made
+* available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution and is available at
+* https://www.eclipse.org/legal/epl-2.0/ or the Apache
+* License, Version 2.0 which accompanies this distribution and
+* is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* This Source Code may also be made available under the
+* following Secondary Licenses when the conditions for such
+* availability set forth in the Eclipse Public License, v. 2.0
+* are satisfied: GNU General Public License, version 2 with
+* the GNU Classpath Exception [1] and GNU General Public
+* License, version 2 with the OpenJDK Assembly Exception [2].
+*
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] https://openjdk.org/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+-->
+
+# -XX:[+|-]EnableExtendedHCR
+
+This option enables or disables the extended Hot Code Replace (HCR) capability in the VM.
+
+## Syntax
+
+        -XX:[+|-]EnableExtendedHCR
+
+| Setting               | Effect  | Default                                                                            |
+|-----------------------|---------|:----------------------------------------------------------------------------------:|
+| `-XX:+EnableExtendedHCR` |  Enable   |  ![Start of content that applies only to Java 8](cr/java8.png)  :fontawesome-solid-check:{: .yes aria-hidden="true"}<span class="sr-only">yes</span>  ![End of content that applies only to Java 8](cr/java_close_lts.png)                                 |
+| `-XX:-EnableExtendedHCR` |  Disable  |  ![Start of content that applies only to Java 11+ (LTS)](cr/java11plus.png)   :fontawesome-solid-check:{: .yes aria-hidden="true"}<span class="sr-only">yes</span>   ![End of content that applies only to Java 11 and later](cr/java_close_lts.png)                       |
+
+## Explanation
+
+Supporting extended HCR has a maintenance cost and because only few users use this capability, the extended HCR feature is deprecated and will be removed in a future release. From OpenJDK 25 onwards, extended HCR will not be supported. Following that, the extended HCR support will be removed from other earlier OpenJDK versions also.
+
+If the extended HCR support is disabled (`-XX:-EnableExtendedHCR`), it might result in a JVMTI failure. In case of such a failure, the following warning message is displayed:
+
+```
+Warning: Extended HCR is disabled, enable with `-XX:+EnableExtendedHCR`. In the future, this capability will be removed from the JDK.
```

Comment:
The message is updated to the following:
Warning: Extended HCR is disabled, enable with -XX:+EnableExtendedHCR. This feature is deprecated. In the future, the extended HCR capability, and the option to enable it, will be completely removed from all versions.